docs/src/user_docs/stdlib/collections.md

```diff
@@ -0,0 +1,22 @@
+# Collections
+Namespace `std::crypto` contains modules for commonly authenticated data structures.
+
+## Merkle Mountain Range
+Module `std::collections::mmr` contains procedures for manipulating [Merkle Mountain Ranges](https://github.com/opentimestamps/opentimestamps-server/blob/master/doc/merkle-mountain-range.md) data structure which can be used as an append-only log.
+
+| Procedure | Description |
+| ----------- | ------------- |
+| get | Loads the leaf at the absolute `pos` in the MMR onto the stack.<br /><br />This MMR implementation supports only u32 positions.<br /><br />Inputs:<br />- Operand stack: [pos, mmr_ptr, ...]<br /><br />Output:<br />- Operand stack: [N, ...]<br /><br />Where `N` is the leaf loaded from the MMR located starting at memory address `mmr_ptr`. |
+| add | Adds a new leaf to the MMR.<br /><br />This will update the MMR peaks in the VM's memory and the advice provider with any merged nodes.<br /><br />Inputs:<br />- Operand stack: [N, mmr_ptr, ...]<br /><br />Outputs:<br />- Operand stack: [...]<br /><br />Where `N` is the leaf added to the MMR located starting at memory address `mmr_ptr`. |
+| pack | Computes the hash of the given MMR and copies it to the Advice Map using its hash as a key.<br /><br />Inputs:<br />- Operand stack: [mmr_ptr, ...]<br /><br />Outputs:<br />- Operand stack: [HASH, ...]<br /><br /> |
+| unpack | Load the MMR peak data based on its hash.<br /><br />Inputs:<br />- Operand stack: [HASH, mmr_ptr, ...]<br /><br />Outputs:<br />- Operand stack: [...]<br /><br />Where:<br />- `HASH`: is the MMR peak hash, the hash is expected to be padded to an even length and to have a minimum size of 16 elements.<br />- The advice map must contain a key with `HASH`, and its value is `num_leaves \|\| hash_data`, and hash_data is the data used to computed `HASH`<br />- `mmt_ptr`: the memory location where the MMR data will be written to, starting with the MMR forest (its total leaves count) followed by its peaks. |
+
+## Sparse Merkle Tree (64)
+
+Module `std::collections::smt64` contains procedures for manipulating key-value maps with single-element keys and 4-element values. Current implementation is a thin wrapper over a simple Sparse Merkle tree of depth 64. In the future, this will be replaced with a compact Sparse Merkle tree implementation.
```

Comment:
nits:
- "Current implementation..." -> "The current implementation..."
- "Sparse Merkle tree" -> "Sparse Merkle Tree". This one actually seems to be done consistently throughout all of the docs below. Is there a reason for that? To me, it seems that it should either be "Sparse Merkle Tree", since that's the name of the data structure, or it should just be "sparse Merkle tree" with only the proper name capitalized

Reply:
That is an excellent point. Should we agree to something? I also use this a lot in the blogposts and docs. I think zkEVM uses Sparse Merkle Tree see [here](https://zkevm.polygon.technology/docs/zkProver/sparse-merkle-tree/).

Reply:
Replaced "Sparse Merkle tree" with "Sparse Merkle Tree" everywhere